libraries/pipes/implementations/rust/src/types_ext.rs

```diff
@@ -38,3 +41,133 @@ impl PipesMessage {
+    fn from(value: String) -> Self {
+        PipesMetadataValue {
+            raw_value: Some(RawValue::String(value)),
+            pipes_metadata_value_type: Some(Type::Text),
+        }
+    }
+}
+
+impl From<HashMap<String, Option<serde_json::Value>>> for PipesMetadataValue {
+    fn from(value: HashMap<String, Option<serde_json::Value>>) -> Self {
+        PipesMetadataValue {
+            raw_value: Some(RawValue::AnythingMap(value)),
+            pipes_metadata_value_type: Some(Type::Json),
+        }
+    }
+}
+
+impl From<Vec<Option<serde_json::Value>>> for PipesMetadataValue {
+    fn from(value: Vec<Option<serde_json::Value>>) -> Self {
+        PipesMetadataValue {
+            raw_value: Some(RawValue::AnythingArray(value)),
+            pipes_metadata_value_type: Some(Type::Json),
+        }
+    }
+}
+
+impl PipesMetadataValue {
+    pub fn new(raw_value: RawValue, pipes_metadata_value_type: Type) -> Self {
+        Self {
+            raw_value: Some(raw_value),
+            pipes_metadata_value_type: Some(pipes_metadata_value_type),
+        }
+    }
+
+    pub fn from_url(url: String) -> Self {
+        PipesMetadataValue {
+            raw_value: Some(RawValue::String(url)),
+            pipes_metadata_value_type: Some(Type::Url),
+        }
+    }
+
+    pub fn from_path(path: String) -> Self {
+        PipesMetadataValue {
+            raw_value: Some(RawValue::String(path)),
+            pipes_metadata_value_type: Some(Type::Path),
+        }
+    }
+
+    pub fn from_notebook(notebook: String) -> Self {
+        PipesMetadataValue {
+            raw_value: Some(RawValue::String(notebook)),
+            pipes_metadata_value_type: Some(Type::Notebook),
+        }
+    }
+
+    pub fn from_md(md: String) -> Self {
+        PipesMetadataValue {
+            raw_value: Some(RawValue::String(md)),
+            pipes_metadata_value_type: Some(Type::Md),
+        }
+    }
+
+    pub fn from_timestamp(timestamp: f64) -> Self {
+        PipesMetadataValue {
+            raw_value: Some(RawValue::Double(timestamp)),
+            pipes_metadata_value_type: Some(Type::Timestamp),
+        }
+    }
+
+    pub fn from_asset(asset: String) -> Self {
+        PipesMetadataValue {
+            raw_value: Some(RawValue::String(asset)),
+            pipes_metadata_value_type: Some(Type::Asset),
+        }
+    }
+
+    pub fn from_job(job: String) -> Self {
+        PipesMetadataValue {
+            raw_value: Some(RawValue::String(job)),
+            pipes_metadata_value_type: Some(Type::Job),
+        }
+    }
+
+    pub fn from_dagster_run(dagster_run: String) -> Self {
+        PipesMetadataValue {
+            raw_value: Some(RawValue::String(dagster_run)),
+            pipes_metadata_value_type: Some(Type::DagsterRun),
+        }
+    }
+
+    pub fn none() -> Self {
+        PipesMetadataValue {
+            raw_value: None,
+            pipes_metadata_value_type: None, // TODO: Should this be `Some(Type::Null)`?
```

Comment:
And should this be `None`, or `Some(Type::Null)`? Not sure what's the difference with either option though.

Reply:
I would go with:
```
    pub fn null() -> Self {
        PipesMetadataValue {
            raw_value: None,
            pipes_metadata_value_type: Some(Type::Null),
```

Function name: `null` is similar to the python implementation [here](https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/pipes/context.py#L163-L164). `Some(Type::Null)` follows convention followed by the other types in the rust implementation

Reply:
Done! Will merge in an hour, if everything's still looks good.

Reply:
LGTM!